digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/TrafficLightService.scala

```diff
@@ -84,21 +92,26 @@ class TrafficLightService(val roadLinkService: RoadLinkService) extends PointAss
     super.getByBoundingBox(user, bounds, roadLinks)
   }
 
-  private def createOperation(asset: PersistedAsset, adjustment: AssetAdjustment): PersistedAsset = {
-    createPersistedAsset(asset, adjustment)
+  override def createOperation(asset: PersistedAsset, adjustment: AssetUpdate): PersistedAsset = {
+    new PersistedAsset(adjustment.assetId, adjustment.linkId, adjustment.lon, adjustment.lat, adjustment.mValue,
+      adjustment.floating, asset.timeStamp, asset.municipalityCode, asset.propertyData, asset.createdBy,
+      asset.createdAt, asset.modifiedBy, asset.modifiedAt, asset.linkSource, asset.externalId)
   }
 
-  private def adjustmentOperation(persistedAsset: PersistedAsset, adjustment: AssetAdjustment, roadLink: RoadLink): Long = {
-    val updated = IncomingTrafficLight(adjustment.lon, adjustment.lat, adjustment.linkId, persistedAsset.propertyData.map(prop => SimplePointAssetProperty(prop.publicId, prop.values)).toSet)
-    updateWithoutTransaction(adjustment.assetId, updated, roadLink, AutoGeneratedUsername.generatedInUpdate,
-                             Some(adjustment.mValue), Some(adjustment.timeStamp))
-  }
-
-  private def createPersistedAsset[T](persistedStop: PersistedAsset, asset: AssetAdjustment) = {
-
-    new PersistedAsset(asset.assetId, asset.linkId, asset.lon, asset.lat,
-      asset.mValue, asset.floating, persistedStop.timeStamp, persistedStop.municipalityCode, persistedStop.propertyData, persistedStop.createdBy,
-      persistedStop.createdAt, persistedStop.modifiedBy, persistedStop.modifiedAt, persistedStop.linkSource, persistedStop.externalId)
+  override def adjustmentOperation(persistedAsset: PersistedAsset, adjustment: AssetUpdate, roadLink: RoadLinkInfo): Long = {
+    val propertyData = persistedAsset.propertyData.map(property =>
+      property.publicId match {
+        case "sidecode" if adjustment.validityDirection.nonEmpty =>
+          val validityDirection = adjustment.validityDirection.get.toString
+          SimplePointAssetProperty(property.publicId, Seq(PropertyValue(validityDirection, Some(validityDirection))))
+        case "bearing" if adjustment.bearing.nonEmpty =>
+          val bearing = adjustment.bearing.get.toString
+          SimplePointAssetProperty(property.publicId, Seq(PropertyValue(bearing, Some(bearing))))
+        case _ => SimplePointAssetProperty(property.publicId, property.values)
+      }).toSet
+    val updated = IncomingTrafficLight(adjustment.lon, adjustment.lat, adjustment.linkId, propertyData)
+    updateWithoutTransaction(adjustment.assetId, updated, roadLink.geometry, roadLink.municipality, AutoGeneratedUsername.generatedInUpdate,
+                             Some(adjustment.mValue), Some(adjustment.timeStamp), LinkGeomSource.NormalLinkInterface)
```

Comment:
```suggestion
                             Some(adjustment.mValue), Some(adjustment.timeStamp), a.linksource)
```
en nyt muista onkos tässä tota kenttää mutta muistelisin että näitä pistemäisiä saatetaan lisätä täydentäviin linkkeihin myös, joten parempi olla käyttämättä LinkGeomSource.NormalLinkInterface

Reply:
Joo no periaatteessa voidaan vaihtaa noinkin. AdjustmentOperationia taidetaan kutsua kyllä ainoastaan samuutuksessa, me ei taideta samuuttaa täydentävällä geometrialla olevia linkkejä?

Reply:
ei taideta, kai voidaan katsoa jos joku päivä halutaan samuutta täydentäviä,
Toisaalta ehkä hyvä periaate vaan käyttää sitä tietoaa mitä asset antaa, kai teoriassa meillä vois tulla linkkejä muista lähteistä myös KGV varaston kautta.

Reply:
Totta, mutta jos käytetään assetin linkSourcea niin oletetaan, että uuden linkin linkSource on sama kuin edellisellä linkillä, johon asset on liitetty. Mut joo voin päivittää tuohon

Reply:
hyym joo pitää sitten sekin sitten päivittä. ei nyt  varmaan kyl akuutti asia muuttaa kun ei nyt tällä hetkellä kyl muualta tule linkkejä